From cb13d6abcebb263859514735a399f7f8b895c27a Mon Sep 17 00:00:00 2001
Message-Id: <cb13d6abcebb263859514735a399f7f8b895c27a.1434058173.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Fri, 8 May 2015 19:44:51 +0530
Subject: [PATCH 1/6] REVERTME: [VPG]: drm/i915: update dl in multipipe case to
 recommended value.

This patch updates display dl value to 0x82 in multipipe
scenario instead of calculated dl. This value is sugested
by hardware team. We need to use this value as the current
dl calculation formula doesnt consider multipipe
scenario.

REVERTME: Once the updated display dl calculation formula
is in use, this patch can be reverted.
Another condition is that, it depends on atomic display
design which would be replaced once nuclear flip is used.

ISSUE: IMINAN-36379
Change-Id: I8f3db5bc0f2a1688521304aad1ab84753c0b671c
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h      |  1 +
 drivers/gpu/drm/i915/intel_display.c | 15 +++++++++++++++
 2 files changed, 16 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index bc58c6a..aaa0b85 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4341,6 +4341,7 @@ enum punit_power_well {
 #define DDL_SPRITEB_PRECISION_H		(1<<23)
 #define DDL_SPRITEB_PRECISION_L		(0<<23)
 #define DDL_SPRITEA_SHIFT	8
+#define DDL_MULTI_PIPE_CHV	0x82
 
 #define DDL_PLANEA_PRECISION_H		(1<<7)
 #define DDL_PLANEA_PRECISION_L		(0<<7)
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 24012eb..74ac3c6 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2740,6 +2740,7 @@ static void i9xx_update_primary_plane(struct drm_crtc *crtc,
 	struct drm_display_mode *mode = &intel_crtc->config.requested_mode;
 	int plane = intel_crtc->plane;
 	int pipe = intel_crtc->pipe;
+	int pipe_stat = VLV_PIPE_STATS(dev_priv->pipe_plane_stat);
 	unsigned long linear_offset;
 	bool rotate = false;
 	bool alpha_changed = false;
@@ -2912,7 +2913,21 @@ static void i9xx_update_primary_plane(struct drm_crtc *crtc,
 	intel_crtc->vlv_wm.sr = vlv_calculate_wm(intel_crtc, pixel_size);
 	plane_ddl = plane_prec_multi | (plane_ddl);
 
+	/*
+	 * The current Dl formula doesnt consider multipipe
+	 * cases, Use this value suggested by sv till the
+	 * actual formula gets used, same applies for all
+	 * hdmi cases. Since secondary display comes on PIPEC
+	 * we are checking for pipe C, pipe_stat variable
+	 * tells us the number of pipes enabled.
+	 */
+	if (IS_CHERRYVIEW(dev))
+		if (!single_pipe_enabled(pipe_stat) ||
+				(pipe_stat & PIPE_ENABLE(PIPE_C)))
+			plane_ddl = DDL_MULTI_PIPE_CHV;
+
 	intel_crtc->reg_ddl.plane_ddl = plane_ddl;
+
 	intel_crtc->reg_ddl.plane_ddl_mask = mask;
 	if (((plane_ddl & mask) != (I915_READ(VLV_DDL(pipe)) & mask)) ||
 			!(dspcntr & DISPLAY_PLANE_ENABLE)) {
-- 
1.9.1

